swarm/api/services/run_state.py

```
@@ -135,16 +135,23 @@ async def update_run(
 
     async def _save_state(self, run_id: str, state: Dict[str, Any]) -> None:
         """Save state to disk and cache."""
-        state_path = self._state_path(run_id)
-        state_path.parent.mkdir(parents=True, exist_ok=True)
 
-        # Write atomically
-        tmp_path = state_path.with_suffix(".tmp")
-        tmp_path.write_text(json.dumps(state, indent=2), encoding="utf-8")
-        os.replace(tmp_path, state_path)
+        def write_state():
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `write_state`

Add return type annotation: `None`

(ANN202)